packages/providers/src/transaction-response/transaction-response.ts

```diff
@@ -199,11 +211,24 @@ export class TransactionResponse {
     const result = await this.waitForResult<TTransactionType>(contractsAbiMap);
 
     if (result.isStatusFailure) {
-      throw new Error(
+      throw new FuelError(
+        ErrorCode.TRANSACTION_FAILED,
         `Transaction failed: ${(<FailureStatus>result.gqlTransaction.status).reason}`
       );
     }
 
     return result;
   }
+
+  /*
+    TODO: Consider adding `maxTimeout` or `maxAttempts` parameter.
+
```

Comment:
```suggestion
    @hidden

    TODO: Consider adding `maxTimeout` or `maxAttempts` parameter.

```

Reply:
@danielbate Is the `@hidden` necessary here?

This block is a multiline comment `/*`, not a DocTag `/**`.

Reply:
@Torres-ssf I am unr-resolving this one, ok? I'm hoping @danielbate can clarify this matter with multiline comments, as I believe it could be problematic if these are treated as DocBlocks.

Reply:
@arboleya ok makes sense

Reply:
Apologies both, you'e correct @arboleya

Reply:
Awesome @danielbate, thanks :pray: